### PR TITLE
Timestamped logging and timings measurement

### PR DESCRIPTION
**Pull request by max-verem**
This pull request (it nest https://github.com/mltframework/mlt/pull/189) makes debugging of some performance depended task much easy - it add timing information to log messages and provides a method for measuring time of execution of some part of code.

Main changes comes in https://github.com/mltframework/mlt/commit/eafb820bc6888ed05e584f2bf63d149e1e295720

### CODE REVIEW

File: src/framework/mlt_log.c [hunk 1]
```diff
@@ -82,3 +101,17 @@ void mlt_log_set_callback( void (*new_callback)( void*, int, const char*, va_lis
...
+	struct timeval tv;
+
+	gettimeofday(&tv, NULL);
+	r = tv.tv_sec;
+
+	r *= 1000000LL;
+	r += tv.tv_usec;
+
+	return r;
+}
```
Comment (ddennedy) [inline src/framework/mlt_log.c]: How do you feel about making mlt_log_timings_now() a noop if NDEBUG is defined?
http://en.cppreference.com/w/c/error/assert

File: src/framework/mlt_log.h [hunk 2]
```diff
@@ -90,4 +91,16 @@ int mlt_log_get_level( void );
 void mlt_log_set_level( int );
 void mlt_log_set_callback( void (*)( void*, int, const char*, va_list ) );
 
+#define mlt_log_timings_begin() \
+{ \
+	int64_t _mlt_log_timings_begin = mlt_log_timings_now(), _mlt_log_timings_end;
+
+#define mlt_log_timings_end(service, msg) \
+	_mlt_log_timings_end = mlt_log_timings_now(); \
+	mlt_log_verbose( service, "%s:%d: T(%s)=%" PRId64 " us\n", \
```
Comment (ddennedy) [inline src/framework/mlt_log.h]: A lot of users or possibly even apps like Kdenlive or Flowblade use verbose log level, and I am not sure we want this volume. OTOH debug level is so verbose it requires text filtering. Maybe add a new level like MLT_LOG_TIMINGS?

**Comment by max-verem** [comment]
> I do not want to use sliced pix_fmt conversion across all versions that we support. I do not like the risk for the bugs that may introduce. Rather, it should be limited to the lower version that @walterav1984 tested and up.

please elaborate

**Comment by ddennedy** [comment]
@walterav1984 said he is using FFmpeg version 2.8.10. So, this has been tested on that version and something very recent by you. Make a build time dependency on [libswscale version 3.1.101 or higher](https://github.com/FFmpeg/FFmpeg/blob/n2.8.10/libswscale/version.h) to use sliced processing with swscale.

**Comment by max-verem** [comment]
> What compiler warning for the lack of space characters? I have not seen that.

```
mlt_slices.c: In function ‘mlt_slices_init’:
mlt_slices.c:201:5: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
     if ( priority < 0 )
     ^~
mlt_slices.c:203:2: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
  pthread_attr_setschedpolicy( &tattr, policy );
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by max-verem** [comment]
> Maybe add a new level like MLT_LOG_TIMINGS?

what about:
```
#define MLT_LOG_TIMINGS 44
```
?

**Comment by max-verem** [comment]
> @walterav1984 said he is using FFmpeg version 2.8.10. So, this has been tested on that version and something very recent by you. Make a build time dependency on libswscale version 3.1.101 or higher to use sliced processing with swscale.

do you mean we have to check minimal version of ffmpeg to use this method and leave original for older version? what checking macro should i use then?

**Comment by walterav1984** [comment]
I can do more tests with newer versions of ffmpeg/chroma upsampling but I / we need to make somekind testing framework with multiple formats / footage and use cases because this incredible great sliced chroma upsampling code may affect alot of use cases. From SD-HD NTSC-PAL 4:2:0 - 4:2:2 playback, field orders, SD downconvertion by melt (HD footage in SD profile) etc etc.

**Comment by ddennedy** [comment]
> do you mean we have to check minimal version of ffmpeg to use this method

Yes

> leave original for older version?

Yes

> what checking macro should i use then?

Really? See all the other library version checking macros used in the same file!

**Comment by ddennedy** [comment]
> #define MLT_LOG_TIMINGS 44

Yes

**Pull request closed without merging**